agent/consul/server.go

```diff
@@ -938,6 +939,23 @@ func isV1CatalogRequest(rpcName string) bool {
 }
 
 func (s *Server) registerControllers(deps Deps, proxyUpdater ProxyUpdater) error {
+	// TODO(CC-6389): Remove once resource-apis is no longer considered experimental and is supported by HCP
+	overrideResourceApisEnabledCheckRaw := os.Getenv("CONSUL_OVERRIDE_HCP_RESOURCE_APIS_CHECK")
+	overrideResourceApisEnabledCheck := false
```

Comment:
Why this is not directly linked to `useV2Resources`? 🤔  or maybe we can add it through another experimental feature flag. I think it's confusing to users to have different ways of configuring features.

Reply:
I think there may be a misunderstanding here. This flag is essentially saying "disable the validation that checks whether v2 resources API is enabled" so it's distinct from `useV2Resources`.

Relevant section of the RFC is [here](https://docs.google.com/document/d/185Kj_pDcuOmmVOzIPZYsCLIIOFB_26bg9kWD7i8qgkQ/edit#heading=h.46rhfevd66e8) and there's also a (now-closed) comment thread on the RFC [here](https://docs.google.com/document/d/185Kj_pDcuOmmVOzIPZYsCLIIOFB_26bg9kWD7i8qgkQ/edit?disco=AAAA8uycqW8).

Essentially, the purpose of this flag is for our own testing and also when we eventually enable Consul clusters using v2 resources to link to HCP, we can tell people who want to try out that functionality early to flip this flag so their cluster is able to link.

All of this being said, if adding an experimental feature flag would be a better way to implement this, that works for me.

Reply:
To add on a little more here, this same environment variable is already used to control this validation when linking via configuration:
https://github.com/hashicorp/consul/blob/9c8e9cebaafd2225f695255c053c54ab6753b048/agent/config/builder.go#L1140-L1147

Reply:
Thank you for the explanation. I think going the feature flag route is better so we can have all our feature flags configured the same way.

Reply:
ok, added a new flag `hcp-v2-resource-apis`. Let me know how that looks!